addons/web_hierarchy/static/src/hierarchy_model.js

```diff
@@ -670,29 +676,71 @@ export class HierarchyModel extends Model {
         config.context = "context" in params ? params.context : config.context;
         if ("domain" in params) {
             config.domain = params.domain;
-            if (!params.domain.length) {
-                if (config.context.hierarchy_res_id) {
-                    config.domain = [["id", "=", config.context.hierarchy_res_id]];
-                    delete config.context.hierarchy_res_id; // just needed for the first load.
-                } else {
-                    config.domain = this.defaultDomain;
+            if (this.isSearchDefaultOrEmpty() && config.context.hierarchy_res_id) {
+                config.domain = [["id", "=", config.context.hierarchy_res_id]];
+                const globalDomain = this.globalDomain;
+                if (globalDomain.length) {
+                    config.domain = Domain.and([config.domain, globalDomain]);
                 }
+                // Just needed for the first load.
+                delete config.context.hierarchy_res_id;
             }
         }
         return config;
     }
 
+    /**
+     * Evaluate if the current search query is the default one.
+     *
+     * @returns {boolean}
+     */
+    isSearchDefaultOrEmpty() {
+        if (!this.env.searchModel) {
+            return true;
+        }
+        const isDisabledOptionalSearchMenuType = (type) => {
+            return (
+                ["filter", "groupBy", "favorite"].includes(type) &&
+                !this.env.searchModel.searchMenuTypes.has(type)
```

Comment:
Yes, in case of an `act_window` with a kanban view and a hierarchy view, you can have `groupBy` filters on the kanban view, then switch to the hierarchy view where `grouBy` filters still exist, can be active, but have no effect (they are active so that they can be recovered when switching back to the kanban)